agentops/instrumentation/google_generativeai/attributes/model.py

```diff
@@ -0,0 +1,274 @@
+        return ""
+
+
+def _set_prompt_attributes(attributes: AttributeMap, args: Tuple, kwargs: Dict[str, Any]) -> None:
+    """Extract and set prompt attributes from the request.
+    
+    Respects privacy controls and handles the various ways prompts can be specified
+    in the Google Generative AI API.
+    
+    Args:
+        attributes: The attribute dictionary to update
+        args: Positional arguments to the method
+        kwargs: Keyword arguments to the method
+    """
+    
+    content = None
+    if args and len(args) > 0:
+        content = args[0]
+    elif 'contents' in kwargs:
+        content = kwargs['contents']
+    elif 'content' in kwargs:
+        content = kwargs['content']
+    
+    if content is None:
+        return
+    
+    if isinstance(content, list):
+        for i, item in enumerate(content):
+            try:
+                extracted_text = _extract_content_from_prompt(item)
+                if extracted_text:
+                    attributes[f"{SpanAttributes.LLM_PROMPTS}.{i}.content"] = extracted_text
+                    role = "user"
+                    if isinstance(item, dict) and "role" in item:
+                        role = item["role"]
+                    elif hasattr(item, "role"):
+                        role = item.role
+                    attributes[f"{SpanAttributes.LLM_PROMPTS}.{i}.role"] = role
+            except Exception as e:
+                logger.debug(f"Error extracting prompt content at index {i}: {e}")
+    else:
+        try:
+            extracted_text = _extract_content_from_prompt(content)
+            if extracted_text:
+                attributes[f"{SpanAttributes.LLM_PROMPTS}.0.content"] = extracted_text
+                attributes[f"{SpanAttributes.LLM_PROMPTS}.0.role"] = "user"
+        except Exception as e:
+            logger.debug(f"Error extracting prompt content: {e}")
+
+
+def _set_response_attributes(attributes: AttributeMap, response: Any) -> None:
+    """Extract and set response attributes from the completion response.
+    
+    Args:
+        attributes: The attribute dictionary to update
+        response: The response from the API
+    """
+    if response is None:
+        return
+    
+    if hasattr(response, "model"):
+        attributes[SpanAttributes.LLM_RESPONSE_MODEL] = response.model
+    
+    if hasattr(response, "usage_metadata"):
+        usage = response.usage_metadata
+        if hasattr(usage, "prompt_token_count"):
+            attributes[SpanAttributes.LLM_USAGE_PROMPT_TOKENS] = usage.prompt_token_count
+        if hasattr(usage, "candidates_token_count"):
+            attributes[SpanAttributes.LLM_USAGE_COMPLETION_TOKENS] = usage.candidates_token_count
+        if hasattr(usage, "total_token_count"):
+            attributes[SpanAttributes.LLM_USAGE_TOTAL_TOKENS] = usage.total_token_count
+    
+    try:
+        if hasattr(response, "text"):
+            attributes[f"{SpanAttributes.LLM_COMPLETIONS}.0.content"] = response.text
```

Comment:
Same here for completions: `MessageAttributes.COMPLETION_ROLE, MessageAttributes.COMPLETION_CONTENT`